mantidproject/mantid
# 38063 sample transmission calculators DoubleSpinBoxes avoid commas

### Description of work
Under `Single` wavelength mode, the spin boxes for `Low`, `Width`, and `High` in the [Sample Transmission Calculator](https://developer.mantidproject.org/Testing/General/SampleTransmissionCalculatorTestGuide.html) interface had allowed its spin boxes to enter values such as 0,2.1 to include a blend of commas and decimal dot and this work is to avoid that.

In addition, this includes fixing two coverity issues in RotateSampleShape.cpp

#### Summary of work




Fixes #38063. 
Fixes #38268 





#### Further detail of work


### To test:
1. Open the Sample Transmission Calculator interface from (Interfaces > General > Sample Transmission Calculator)
2. Enter a value containing comma in the DoubleSpinBoxes for any of `Low`, `Width`, and `High` fields ex: 2,3
3. The moment when the number following the comma is entered the the comma should be converted to a decimal dot.
4. Try entering different combinations of dot and comma and the same should be observed as in step 3 







---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
